streaming/base/converters/dataframe_to_mds.py

```diff
@@ -261,6 +262,9 @@ def write_mds(iterator: Iterable):
     if cu.remote is None:
         mds_path = (cu.local, '')
     else:
+        if dataframe.isStreaming:
```

Comment:
Not sure why only outputting to local is possible here? What is the challenge to support original ``out`` (meaning local, remote or (local, remote))?

Reply:
We want to disallow concurrent writes to the top-level index.json. We need a lock file that is shared among all workers. This is implemented here using open with flags to ensure that opening the lock file fails if it already exists.